RocketChat/Rocket.Chat.ReactNative
# [FIX] Cancelling when adding a server

## Proposed changes


Disable the close button while connecting and enable it again in case of an error

## Issue(s)	


## How to test or reproduce


When opening the Servers List, on Room List View, the user has the option to add a new server.

If they click on it, they are redirected to a screen that is just like the "first screen of the app", on which the user types in the server they want and is moved on to continue to the login process.

This can also happen If the user taps on a server that's listed but has no user logged in.

However, this screen, unlike the "original first screen", can be closed by an "x" icon on the top left.

If this screen is closed before the app finishes "loading" the choice of server and redirects the user to the next screen, the app brings the user to the server they came from at the start of the process, but after a few seconds, redirects the user to the "actual" first screen of the app. One with no "x" to be closed. Effectively kicking the user out of logged servers and forcing the user to enter workspaces again.

It's fair to note that they are not "logged off". If they type a server that they were logged into, they don't need to go through the "log in" process again.

This is fixed by restarting the app.

## Screenshots

https://user-images.githubusercontent.com/47038980/192863934-98ff20aa-ed0e-4791-a27e-bf1e49204b19.mov


## Types of changes




## Checklist



## Further comments


Jira: [TC-110]


[TC-110]: https://rocketchat.atlassian.net/browse/TC-110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
